src/main/java/com/elasticpath/tools/mavenminimal/reactor/MavenDriver.java

```diff
@@ -145,6 +147,28 @@ private static CommandLine determineMavenCommand(final Reactor reactor, final Li
 		return cmdLine;
 	}
 
+
+	/**
+	 * The number of threads can be specified in several ways for Maven, each possibility needs to be removed.
+	 * This method
+	 * @param mavenArguments the arguments to pass to maven.
+	 */
+	static void removeThreadingFlags(final List<String> mavenArguments) {
+
+		mavenArguments.removeIf(s -> s != null && s.matches("(-T.+.*|--threads=.*)"));
+
+		for (Iterator<String> iter = mavenArguments.iterator(); iter.hasNext();) {
+			String arg = iter.next();
+			if (arg.matches("(-T|--threads)")) {
+				iter.remove();          // Remove the flag
+				if (iter.hasNext()) {   // Remove the argument
+					iter.next();
+					iter.remove();
+				}
+			}
+		}
```

Comment:
Isn't this iterator doing exactly the same thing as the "removeIf" on line 158?

Reply:
No, 158 gets rid of the flags with have spaces: `-T4` or `--threads=4`
This iterator deals with flags that are provided with a space (and therefore two list elements): `-T 4` or `--threads 4`

Reply:
I guess not exactly, but couldn't the "removeIf" be updated to "(-T.+.*|--threads=.*|-T|--threads)"?

Reply:
See my new comment above.